x/protocolpool/keeper/keeper.go

```diff
@@ -201,46 +187,52 @@ func (k Keeper) SetToDistribute(ctx context.Context, amount sdk.Coins, addr stri
 		return err
 	}
 
-	totalStreamFundsPercentage := math.ZeroInt()
-	err = k.RecipientFundPercentage.Walk(ctx, nil, func(key sdk.AccAddress, value math.Int) (stop bool, err error) {
-		totalStreamFundsPercentage = totalStreamFundsPercentage.Add(value)
+	totalStreamFundsPercentage := math.LegacyZeroDec()
+	err = k.ContinuousFund.Walk(ctx, nil, func(key sdk.AccAddress, cf types.ContinuousFund) (stop bool, err error) {
+		// Check if the continuous fund has expired
+		if cf.Expiry != nil && cf.Expiry.Before(k.HeaderService.HeaderInfo(ctx).Time) {
+			return false, nil
+		}
+
+		totalStreamFundsPercentage = totalStreamFundsPercentage.Add(cf.Percentage)
+		if totalStreamFundsPercentage.GT(math.LegacyOneDec()) {
+			return true, fmt.Errorf("total funds percentage cannot exceed 100")
+		}
+
 		return false, nil
 	})
-	if totalStreamFundsPercentage.GT(math.NewInt(100)) {
-		return fmt.Errorf("total funds percentage cannot exceed 100")
-	}
 	if err != nil {
 		return err
 	}
 
+	// if percentage is 0 then return early
+	if totalStreamFundsPercentage.IsZero() {
+		return nil
+	}
+
 	// send streaming funds to the stream module account
-	if err := k.sendFundsToStreamModule(ctx, denom, totalStreamFundsPercentage); err != nil {
+	toDistributeDec := sdk.NewDecCoin(denom, amount.AmountOf(denom))
+	if totalStreamFundsPercentage.LT(math.LegacyOneDec()) {
```

Comment:
nit i dont think we need to add a check here, since the above check already return an error if percentage > 100

Reply:
yeah this is kind of a stupid "optimization", that in case the entire funds where being distributed we would avoid doing some extra math. I'll simplify this